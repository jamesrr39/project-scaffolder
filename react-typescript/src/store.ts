import { configureStore } from "@reduxjs/toolkit";
import { setupListeners } from "@reduxjs/toolkit/query";
import counterReducer from "./counter";

export const store = configureStore({
  reducer: {
    counter: counterReducer,
  },
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;

// see `setupListeners` docs - takes an optional callback as the 2nd arg for customization
setupListeners(store.dispatch);
