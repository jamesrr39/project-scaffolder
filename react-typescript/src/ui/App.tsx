import { useDispatch, useSelector } from "react-redux";
import { add } from "src/counter";
import { RootState } from "src/store";

function App() {
  const count = useSelector((state: RootState) => {
    return state.counter.count;
  });

  const dispatch = useDispatch();

  return (
    <div>
      <p>Current count: {count}</p>
      <button
        type="button"
        onClick={() => {
          dispatch(add({ amount: 1 }));
        }}
      >
        Add one
      </button>
    </div>
  );
}

export default App;
