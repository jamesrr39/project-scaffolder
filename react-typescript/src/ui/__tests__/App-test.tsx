import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import App from "../App";
import { store } from "../../store";

describe("App", () => {
  test("render", () => {
    const component = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
