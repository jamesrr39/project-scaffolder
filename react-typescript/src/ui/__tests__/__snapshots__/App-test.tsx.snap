// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render 1`] = `
<div>
  <p>
    Current count: 
    0
  </p>
  <button
    onClick={[Function]}
    type="button"
  >
    Add one
  </button>
</div>
`;
